server/src/main/java/org/apache/iotdb/db/metadata/schemaregion/SchemaRegionMemoryImpl.java

```diff
@@ -409,6 +413,72 @@ public synchronized void deleteSchemaRegion() throws MetadataException {
     SchemaRegionUtils.deleteSchemaRegionFolder(schemaRegionDirPath, logger);
   }
 
+  @Override
+  public synchronized boolean createSnapshot(File snapshotDir) {
+    File mLogSnapshotTmp =
+        SystemFileFactory.INSTANCE.getFile(snapshotDir, MetadataConstant.METADATA_LOG_SNAPSHOT_TMP);
+    File mLogSnapshot =
+        SystemFileFactory.INSTANCE.getFile(snapshotDir, MetadataConstant.METADATA_LOG_SNAPSHOT);
+
+    try {
+      logWriter.copyTo(mLogSnapshotTmp);
+      mLogSnapshot.delete();
+      if (!mLogSnapshotTmp.renameTo(mLogSnapshot)) {
+        logger.error(
+            "Failed to rename {} to {} while creating snapshot for schemaRegion {}.",
+            mLogSnapshotTmp.getName(),
+            mLogSnapshot.getName(),
+            schemaRegionId);
+        mLogSnapshot.delete();
```

Comment:
Why is `delete()` invoked again here ?

Reply:
I'm not sure whether the ```mLogSnapshot``` will be produced if the rename operation is failed. If not, the invoking is unnecessary.